examples/lock-app/lock-common/src/LockEndpoint.cpp

```diff
@@ -18,29 +18,48 @@
 #include "LockEndpoint.h"
 #include <app-common/zap-generated/attributes/Accessors.h>
 #include <cstring>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/internal/CHIPDeviceLayerInternal.h>
 
 using chip::to_underlying;
 using chip::app::DataModel::MakeNullable;
 
-bool LockEndpoint::Lock(const Optional<chip::ByteSpan> & pin, OperationErrorEnum & err, OperationSourceEnum opSource)
+struct LockActionData
 {
-    return setLockState(DlLockState::kLocked, pin, err, opSource);
+    chip::EndpointId endpointId;
+    DlLockState lockState;
+    OperationSourceEnum opSource;
+    Nullable<uint16_t> userIndex;
+    uint16_t credentialIndex;
+    Nullable<chip::FabricIndex> fabricIdx;
+    Nullable<chip::NodeId> nodeId;
+    bool moving = false;
+};
+
+static LockActionData gCurrentAction;
+
+bool LockEndpoint::Lock(const Nullable<chip::FabricIndex> & fabricIdx, const Nullable<chip::NodeId> & nodeId,
+                        const Optional<chip::ByteSpan> & pin, OperationErrorEnum & err, OperationSourceEnum opSource)
+{
+    return setLockState(fabricIdx, nodeId, DlLockState::kLocked, pin, err, opSource);
 }
 
-bool LockEndpoint::Unlock(const Optional<chip::ByteSpan> & pin, OperationErrorEnum & err, OperationSourceEnum opSource)
+bool LockEndpoint::Unlock(const Nullable<chip::FabricIndex> & fabricIdx, const Nullable<chip::NodeId> & nodeId,
```

Comment:
Possibly a silly question (not my normal area), but what would a null fabricIdx mean here? Unlock over PASE? Is that allowed?

Reply:
On a real lock, a null fabric index could occur when the action source is not via Matter (for example, locking via thumb turn)

Reply:
AH. Cool thanks.